Sources/ScheduledTasks.php

```diff
@@ -1246,12 +1246,16 @@ function scheduled_fetchSMfiles()
 	foreach ($js_files as $ID_FILE => $file)
 	{
 		// Create the url
-		$server = empty($file['path']) || (substr($file['path'], 0, 7) != 'http://' && substr($file['path'], 0, 8) != 'https://') ? 'http://www.simplemachines.org' : '';
+		$server = empty($file['path']) || (substr($file['path'], 0, 7) != 'http://' && substr($file['path'], 0, 8) != 'https://') ? 'https://www.simplemachines.org' : '';
 		$url = $server . (!empty($file['path']) ? $file['path'] : $file['path']) . $file['filename'] . (!empty($file['parameters']) ? '?' . $file['parameters'] : '');
 
 		// Get the file
 		$file_data = fetch_web_data($url);
 
+		// If we got an error - maybe https isn't working, try http?
+		if ($file_data === false && substr($url, 0, 30) == 'https://www.simplemachines.org')
+			$file_data = fetch_web_data(str_replace('https://www.simplemachines.org', 'http://www.simplemachines.org', $url));
```

Comment:
There isn't any built in method currently for our system to check the ssl certs.  The fallback is to currently allow the system to fall back incase port 443 or something else (like a weird php configuration I don't know about) prevents being able to fetch data over 443.  I thought about taking out http completely but wasn't sure.

Thinking about it if they don't have direct access to the machine, this would only occur with dns poisoning.  If they had direct access to the machine, they could change the DNS, but then also inject a new trusted root ca and self sign a certificate that would be trusted as well. There isn't stuff yet in the code to check for HSTS or similar either.